### DISCUSSION TITLE
Added: C# Language Design Notes for May 17, 2017

### DISCUSSION THREAD

**Discussion by MadsTorgersen** [dotnet/csharplang#689] [General]
[C# Language Design Notes for May 17, 2017](https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-05-17.md)

More questions about default interface member implementations

1. Conflicting override of default implementations
2. Can the Main entry point method be in an interface?
3. Static constructors in interfaces?
4. Virtual properties with private accessors
5. Does an override introduce a member?
6. Parameter names

Please discuss.

**Comment by gulshan** [comment]
👍 for static constructors in interfaces. But #169 may be another way to enable this.

**Comment by alrz** [comment]
>Conflicting override of default implementations

But we already do pick an arbitrary impl (first in base list) in case of [multiple interface implementations](https://stackoverflow.com/questions/14562047/generic-type-parameter-covariance-and-multiple-interface-implementations) involving covariance,

> ```cs
> void Main()
> {
> 	IGeneric<Base> b = new DoubleDown();
>         // prints "Derived1" or "Derived2" depending on the base order
> 	Console.WriteLine(b.GetName()); 
> }
> 
> interface IGeneric<out T>
> {
>     string GetName();
> }
> 
> class Base {}
> class Derived1 : Base{}
> class Derived2 : Base{}
> 
> class DoubleDown : IGeneric<Derived1>, IGeneric<Derived1> // order is important
> {
>    string IGeneric<Derived1>.GetName()
>    {
>      return "Derived1";
>    }
> 
>    string IGeneric<Derived2>.GetName()
>    {
>      return "Derived2";
>    }  
> }
> ```
>
